src/HttpMock/EndpointMatchingRule.cs

```diff
@@ -28,7 +29,14 @@ public class EndpointMatchingRule : IMatchingRule
 				queryParamMatch = new QueryParamMatch().MatchQueryParams(requestHandler, requestQueryParams);
 			}
 
-			return uriStartsWith && httpMethodsMatch && queryParamMatch;
+			bool headerMatch = true;
+			bool shouldMatchHeaders = (requestHandler.RequestHeaders.Count > 0);
+
+			if (shouldMatchHeaders) {
+				headerMatch = new HeaderMatch().MatchHeaders(requestHandler, requestHeaders);

```

Comment:
Make this a field, and initialise it in the ctor. Planning on making testing a bit easier for this class.